test/test_response.py

```diff
@@ -254,6 +254,38 @@ def test_decode_brotli_error(self) -> None:
         with pytest.raises(DecodeError):
             HTTPResponse(fp, headers={"content-encoding": "br"})
 
+    @onlyZstd()
+    def test_decode_zstd(self) -> None:
+        data = zstd.compress(b"foo")
+
+        fp = BytesIO(data)
+        r = HTTPResponse(fp, headers={"content-encoding": "zstd"})
+        assert r.data == b"foo"
+
+    @onlyZstd()
+    def test_chunked_decoding_zstd(self) -> None:
+        data = zstd.compress(b"foobarbaz")
+
+        fp = BytesIO(data)
+        r = HTTPResponse(
+            fp, headers={"content-encoding": "zstd"}, preload_content=False
+        )
+
+        ret = b""
+
+        for _ in range(100):
+            ret += r.read(1)
+            if r.closed:
+                break
+        assert ret == b"foobarbaz"
+
+    @onlyZstd()
+    def test_decode_zstd_error(self) -> None:
+        fp = BytesIO(b"fooooooooooooooo")
+
+        with pytest.raises(DecodeError):
+            HTTPResponse(fp, headers={"content-encoding": "zstd"})
+
```

Comment:
Can we use pytest.mark.paramettrize with marks as done in `test_make_headers` to avoid duplicating those tests a fourth time? (Could be a follow-up PR if you prefer.)

Reply:
Ok, for a follow-up.